### PR TITLE
[TEVA-1316] svg consistency

### PR DESCRIPTION
**Pull request by alexbowen**
https://dfedigital.atlassian.net/browse/TEVA-1316

Change all use of svgs to be using `` tag in markup or background-images in CSS to load associated SVG asset.

This was deemed the most appropriate for application needs and based on this comparison:

`` | `<object>` | Inline SVG

Browser support | Good | Good | Good
Alt and title attribute | Yes | None | Title only
Browser caching | Yes | Yes | None
GZip compression | Yes | Yes | None
Interactivity | None | Yes | Very good
Search engine indexing | Yes | Through fallback | None
Workflow | Streamlined | Medium | Convoluted
Loading speed | Fast | Slower | Very fast
Maintenance | Easy | Easy | Difficult

using the img tag has the draw back that you cant interactively style them, hence the need for a color variant with the alert icon.

- remove usage of font awesome as was only used in one place, so one less CDN to connect to
- refactor affected styles and markup to use new images
- remove fallback images as not needed due to browser support
- updated social media buttons to more modern versions (okay'd with chris)

### CODE REVIEW

**Comment by alexbowen** [comment]
i am a bit stumped why these tests are failing as they pass for me locally and i cant see that any changes i have made affect these tests  :| anyone have any ideas?

File: app/components/jobseekers/vacancy_details_component.html.haml [hunk 1]
```diff
@@ -37,10 +37,6 @@
       .supporting-docs
         %ul
           - @vacancy.documents.each do |document|
-            %li
-              .document-link
-                %svg{xmlns:"http://www.w3.org/2000/svg", width:"12", height:"18", viewBox:"0 0 16 22", fill:"none"}
-                  %path{d:"M2 0C1.46957 0 0.960859 0.210714 0.585786 0.585786C0.210714 0.960859 0 1.46957 0 2V18C0 18.5304 0.210714 19.0391 0.585786 19.4142C0.960859 19.7893 1.46957 20 2 20H14C14.5304 20 15.0391 19.7893 15.4142 19.4142C15.7893 19.0391 16 18.5304 16 18V6L10 0H2ZM10 7H9V2L14 7H10Z", fill:"#0b0c0c"}
-                    %image{src: asset_pack_path("media/images/icon-document.png")}
+            %li.icon.icon--left.icon--document
```
Comment (cpjmcquillan) [inline app/components/jobseekers/vacancy_details_component.html.haml:40]: This is a bit out of line now

**Review by csutter — APPROVED** [review]
Looks great 👍 

Do we need `app/frontend/src/styles/application/README.md`?

**Review by cesidio — APPROVED** [review]

**Comment by alexbowen** [comment]
> Looks great 👍
> 
> Do we need `app/frontend/src/styles/application/README.md`?

you are absolutley right, i did start one. ill finish and put it in seperate PR. once this is intergrated with stuff cesi has done there is more to document anyway

**Pull request merged**